Neos.Flow/Classes/Mvc/ActionResponse.php

```diff
@@ -264,4 +265,52 @@ public function mergeIntoComponentContext(ComponentContext $componentContext): C
 
         return $componentContext;
     }
+
+    /**
+     * Note this is a special use case method that will apply the internal properties (Content-Type, StatusCode, Location, Set-Cookie and Content)
+     * to the given PSR-7 Response and return a modified response. This is used to merge the ActionResponse properties into a possible HttpResponse
+     * created in a View (see ActionController::renderView()) because those would be overwritten otherwise. Note that any component parameters will
+     * still run through the component chain and will not be propagated here.
+     *
+     * WARNING: Should this ActionResponse contain body content it would replace any content in the given HttpReponse.
+     *
+     * @param ResponseInterface $httpResponse
+     * @return ResponseInterface
+     * @internal
+     */
+    public function applyToHttpResponse(ResponseInterface $httpResponse): ResponseInterface
+    {
+        if ($this->statusCode !== 200) {
```

Comment:
IMO we should change the default to `null` here, check for that and only merge the status if it has been set explicitly

Reply:
I am not sure if that wasn't a breaking change now that we have the default at 200, but I agree to the sentiment. I definitely do not want to do that in this change...

Reply:
Fair enough and I don't want to add even more work. But FYI since the class is final it would be 100% b/c if we'd change
```php
    public function getStatusCode(): int
    {
        return $this->statusCode ?? 200;
    }
```

Reply:
Right, that makes sense I guess, seems a bit hidden tricky but I take it ;)

Reply:
Ah BUT, that actually has another problem... Similar to the content problem. The plugins and widgets WILL probably always set a status code to the response given that it's a full ActionController run, now if the plugin returned 200 it could technically overwrite a redirect (or any other status code in the HTTP response) which would then be unexpected...

Reply:
> The plugins and widgets WILL probably always set a status code

Is that so? They should only set that if it's explicitly set (i.e. by a redirect or via `$this->response->setStatusCode()`). If that's not the case, it's probably too big of a change..

Reply:
You are right, we apparently never set it unless we must :D I expected worse. I guess I will do it and we see where that leaves us.